internal/driven/storage/mysql/userstr/storage.go

```diff
@@ -39,9 +38,15 @@ func (s *Storage) RegisterNewUser(ctx context.Context, user *entity.User) (err e
 	return nil
 }
 
-func (s *Storage) LoginUser(ctx context.Context, req request.LoginRequestBody) (user entity.User, err error) {
-	// implement your logic here
-	return entity.User{}, nil
+func (s *Storage) LoginUser(ctx context.Context, user *entity.User) (err error) {
+	// note: this is not a good practice to store password in plain text but this is a requirement
+	query := `SELECT * FROM users WHERE username = ? AND password = ?`
+	err = s.sqlClient.GetContext(ctx, user, query, user.Username, user.Password)
+	if err != nil {
+		return fmt.Errorf("unable to execute query: %w", err)
```

Comment:
Handle sql err no rows properly. SQL Err no rows is error thrown by MySQL when there's no row meet the query condition in the db. Check [this](https://github.com/Haraj-backend/hex-monscape/blob/51cea85bdfd83efb347e3ec4dbcd6ebe193ec70f/internal/driven/storage/mysql/monstrg/storage.go#L65) as ref.